bin/reth/src/commands/stage/unwind.rs

```diff
@@ -59,25 +61,35 @@ impl Command {
             eyre::bail!("Database {db_path:?} does not exist.")
         }
 
-        let db =
-            open_db(db_path.as_ref(), DatabaseArguments::default().log_level(self.db.log_level))?;
+        let db = Arc::new(open_db(
+            db_path.as_ref(),
+            DatabaseArguments::default().log_level(self.db.log_level),
+        )?);
 
         let range = self.command.unwind_range(&db)?;
 
         if *range.start() == 0 {
             eyre::bail!("Cannot unwind genesis block")
         }
 
-        let factory = ProviderFactory::new(&db, self.chain.clone(), data_dir.static_files_path())?;
-        let provider = factory.provider_rw()?;
+        let provider_factory =
+            ProviderFactory::new(db.clone(), self.chain.clone(), data_dir.static_files_path())?;
 
-        let blocks_and_execution = provider
-            .take_block_and_execution_range(&self.chain, range)
-            .map_err(|err| eyre::eyre!("Transaction error on unwind: {err}"))?;
+        let pipeline_builder = Pipeline::builder();
```

Comment:
i think it builds an empty pipeline with no stages, but we need to add a set of default stages to replicate the actual pipeline unwind

Reply:
Does this mean that we have to implement what is implemented in `take_block_and_execution_range` in pipeline stages?

Reply:
we might just need to use `DefaultStages`, I would check out how reth builds a pipeline normally and try to replicate that here

Reply:
Quick question. It seems I need a `TaskExecutor` in order to create a header and a body downloader for the `DefaultStages`. Is it fine to initialize in this function an `AsyncCliRunner` or I can should I pass it from a higher level? Thanks in advance

Reply:
you can see the example of initiating `DefaultStages` in https://github.com/paradigmxyz/reth/blob/main/bin/reth/src/commands/import.rs

Reply:
Import pipeline is using a `FileClient` to build the header and body downloaders. 
What type of client should I use?

Reply:
> Import pipeline is using a `FileClient` to build the header and body downloaders. What type of client should I use?

The `FetchClient` should be used, although this tells me that we should have some sort of pipeline unwind method that does not require creating a network